nntrainer/tensor/tensor_base.cpp

```diff
@@ -342,4 +342,236 @@ void TensorBase::calculateFlattenDot(
   ldc = (getFormat() == Tformat::NHWC) ? output.channel() : output.width();
 }
 
+/**
+ * Please note that the following functions need to be implemented in a child
+ * class to utilize tensor operations fully — operations such as addition,
+ * division, multiplication, dot production, data averaging, and so on.
+ */
+void TensorBase::setRandNormal(float mean, float stddev) {
+  throw std::invalid_argument(
+    "Tensor::setRandNormal() is currently not supported in tensor data type " +
+    getStringDataType());
+}
+
+void TensorBase::setRandUniform(float min, float max) {
+  throw std::invalid_argument(
+    "Tensor::setRandUniform() is currently not supported in tensor data type " +
+    getStringDataType());
+}
+
+void TensorBase::setRandBernoulli(float probability) {
+  throw std::invalid_argument("Tensor::setRandBernoulli() is currently not "
+                              "supported in tensor data type " +
+                              getStringDataType());
+}
+
+Tensor TensorBase::multiply_strided(Tensor const &m, Tensor &output,
+                                    const float beta) const {
+  throw std::invalid_argument("Tensor::multiply_strided() is currently not "
+                              "supported in tensor data type " +
+                              getStringDataType());
+  return output;
+}
+
+int TensorBase::multiply_i(float const &value) {
+  throw std::invalid_argument(
+    "Tensor::multiply_i() is currently not supported in tensor data type " +
+    getStringDataType());
+  return ML_ERROR_NOT_SUPPORTED;
+}
+
+Tensor &TensorBase::multiply(float const &value, Tensor &output) const {
+  throw std::invalid_argument(
+    "Tensor::multiply() is currently not supported in tensor data type " +
+    getStringDataType());
+  return output;
+}
+
```

Comment:
not really. If no overridden function exists in the derived class, it will throw an error.
On top of that, you must implement any mathematical tensor operations in the derived class (e.g., FloatTensor).